### PR TITLE
feat(type-safe-api): faster code generation for java runtime

### PR DESCRIPTION
**Pull request by cogwirrel**
Move java runtime to new codegen.

Note that we no longer generate markdown docs as part of the Java runtime, but users can generate similar markdown docs in the dedicated documentation project.

### CODE REVIEW

File: packages/type-safe-api/scripts/type-safe-api/generators/generate-next.ts [hunk 1]
```diff
@@ -178,6 +178,40 @@ const toPythonName = (namedEntity: 'model' | 'property' | 'operation', name: str
   return nameSnakeCase;
 };
```
Comment (martenpayne) [inline packages/type-safe-api/scripts/type-safe-api/generators/generate-next.ts:180]: is the list below complete against https://en.wikipedia.org/wiki/List_of_Java_keywords ?

File: packages/type-safe-api/scripts/type-safe-api/generators/java/templates/client/auth/apiKeyAuth.ejs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  "overwrite": true
+}
+###/TSAPI_WRITE_FILE###<%- include('../../header.partial.ejs', { info }) %>
+
+
+package <%- metadata.packageName %>.auth;
+
+import <%- metadata.packageName %>.ApiException;
+import <%- metadata.packageName %>.Pair;
+
```
Comment (martenpayne) [inline packages/type-safe-api/scripts/type-safe-api/generators/java/templates/client/auth/apiKeyAuth.ejs:15]: can the bulk of these Java module templates be read in as a data resource at runtime (possibly allowing for some level of validation with javac)?

File: packages/type-safe-api/scripts/type-safe-api/generators/java/templates/server/operationConfig.ejs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// Generic type for object "keyed" by operation names
+@lombok.Builder @lombok.Getter
+public class OperationConfig<T> {
+    <%_ allOperations.forEach((operation) => { _%>
+    private T <%- operation.name %>;
+    <%_ }); _%>
+
+    public Map<String, T> asMap() {
+        Map<String, T> map = new HashMap<>();
```
Comment (martenpayne) [inline packages/type-safe-api/scripts/type-safe-api/generators/java/templates/server/operationConfig.ejs:24]: I'm a little rusty on Java generics; is this <> syntax valid?

**Review by martenpayne — COMMENTED** [review]
Just left a couple questions/comments…

File: packages/type-safe-api/scripts/type-safe-api/generators/generate-next.ts (hunk 1 above)
Comment (cogwirrel) [reply packages/type-safe-api/scripts/type-safe-api/generators/generate-next.ts:180]: It's the same as OpenAPI generator's here for parity :) https://github.com/OpenAPITools/openapi-generator/blob/8f2676c5c2bcbcc41942307e5c8648cee38bcc44/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java#L179

I'll add a comment to the code to link it!

File: packages/type-safe-api/scripts/type-safe-api/generators/java/templates/server/operationConfig.ejs (hunk 3 above)
Comment (cogwirrel) [reply packages/type-safe-api/scripts/type-safe-api/generators/java/templates/server/operationConfig.ejs:24]: [Yes!](https://docs.oracle.com/javase/tutorial/java/generics/types.html?ref=sangkon.com#:~:text=In%20Java%20SE%207%20and,type%20arguments%20from%20the%20context.) 😄

File: packages/type-safe-api/scripts/type-safe-api/generators/java/templates/client/auth/apiKeyAuth.ejs (hunk 2 above)
Comment (cogwirrel) [reply packages/type-safe-api/scripts/type-safe-api/generators/java/templates/client/auth/apiKeyAuth.ejs:15]: I think compiling the generated code as part of some kind of integration test would be the way to go here - I've raised https://github.com/aws/aws-pdk/issues/868 to track that :)

**Review by martenpayne — APPROVED** [review]

**Pull request merged**